### PR TITLE
detect scale diff

### PR DESCRIPTION
**Pull request by zyndiecate**
TBD

### CODE REVIEW

File: diff.go [hunk 1]
```diff
@@ -53,12 +53,25 @@ const (
...
 
-	// DiffInfoComponentScaleUpdated
-	DiffInfoComponentScaleUpdated DiffType = "component-scale-updated"
+	// scale
+
+	// DiffInfoComponentScalePlacementUpdated
+	DiffInfoComponentScalePlacementUpdated DiffType = "component-scale-placement-updated"
+
+	// DiffInfoComponentScaleUp
+	DiffInfoComponentScaleUp DiffType = "component-scale-up"
```
Comment (ewoutp) [inline diff.go]: Why is there a `component-max-updated` and not a `component-min-updated`?

PS. I think both should be named `component-scale-mXX-updated`

Comment (zyndiecate) [inline diff.go]: I agree with the naming. Will fix it. 

Reason for not having `component-min-updated` is that we want to distinguish in a more detailed way what exactly was updated in min. So we can evaluate if we want to scale up or scale down.

Comment (ewoutp) [inline diff.go]: Not sure this is always sufficient.

What about the following situation:

- component has scale min: 1, max: 3
- component has 3 instances

now you submit a new definition that has scale min: 1, max: 2
this should result in a scale-down also.

Comment (zyndiecate) [inline diff.go]: This is a great issue. Will think about it. This actually requires knowledge of the current environment. :confused:

File: diff.go [hunk 2]
```diff
@@ -380,33 +391,95 @@ func diffComponentExpose(oldDef, newDef ComponentDefinition) []DiffInfo {
...
-func diffComponentScale(oldDef, newDef ComponentDefinition) []DiffInfo {
+// diffComponentScale checks in detail what diff type should be applied between
+// oldDef and newDef. The following can be applied.
+//   placement changed -> DiffTypeComponentScalePlacementUpdated
+//   min decreased     -> DiffTypeComponentScaleMinDecreased
+//   min increased     -> DiffTypeComponentScaleMinIncreased
+//   max decreased     -> DiffTypeComponentScaleMaxDecreased
+//   max increased     -> DiffTypeComponentScaleMaxIncreased
+func diffComponentScale(oldDef, newDef ComponentDefinition, componentName ComponentName) []DiffInfo {
+	oldScaleDef := &ScaleDefinition{}
```
Comment (ewoutp) [inline diff.go]: This does not take into account default values. This has already been noted in app-service, so we'll deal with it later.

**Comment by ewoutp** [comment]
LGTM & WFM

File: diff.go (hunk 2 above)
Comment (zyndiecate) [inline diff.go]: Not sure. This is just about having diffs. The code is doing that. What we would need to do is to default given definitions before creating the diff. As you said, we need to do this in a later step.

**Pull request merged**